### PR TITLE
Feature/5835 base variable

### PR DESCRIPTION
**Pull request by saminahbab**
### What

Add endpoint for base variable given categorization to population types api 

add component tests 

PLEASE NOTE: no swagger spec was found for the endpoint despite link and so did best guess. 
ALSO: The cantabular server returns inconsistent error messages and given the way the api client is built, some not found codes are actually returned as bad gateway. This is left as is for now. 

### How to review

run make test and make test-component 

### Who can review

team-b

### CODE REVIEW

File: handler/dimensions.go [hunk 1]
```diff
@@ -174,6 +173,64 @@ func (h *Dimensions) GetCategorisations(w http.ResponseWriter, r *http.Request)
 	h.respond.JSON(ctx, w, http.StatusOK, resp)
 }
 
+//  GetBase returns the base variables for the
+func (h *Dimensions) GetBase(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (saminahbab) [reply handler/dimensions.go]: Jit that should be sorted now

File: handler/dimensions.go [hunk 2]
```diff
@@ -174,6 +173,64 @@ func (h *Dimensions) GetCategorisations(w http.ResponseWriter, r *http.Request)
...
+			message: "failed to get base variable",
+			logData: logData,
+		})
+		return
+	}
+
+	resp := contract.GetBaseVariableResponse{}
+
+	if res != nil {
+		resp.Name = res.Dataset.Variables.Edges[0].Node.MapFrom[0].Edges[0].Node.Name
```
[the remark this answers is not in the document]
Comment (saminahbab) [reply handler/dimensions.go:235]: The nil pointer should not happen as the client errors on request, not returning empty lists but added a check

File: features/mock/cantabular_client.go [hunk 3]
```diff
@@ -175,3 +179,31 @@ func (c *CantabularClient) GetCategorisations(_ context.Context, _ cantabular.Ge
 	}
 	return c.GetCategorisationsResponse, nil
 }
+
+func (c *CantabularClient) GetBaseVariable(_ context.Context, _ cantabular.GetBaseVariableRequest) (*cantabular.GetBaseVariableResponse, error) {
+	if !c.Healthy {
+		return nil, dperrors.New(
+			errors.New("test error response"),
```
Comment (j-s-rawat) [inline features/mock/cantabular_client.go:186]: found this in other places in the file too, so not a must fix, but seems the error message is not useful at all

**Review by j-s-rawat — APPROVED** [review]
LGTM

**Pull request merged**